ern-core/src/FsCache.ts

```diff
@@ -0,0 +1,220 @@
+import fs from 'fs'
+import path from 'path'
+import { readJSON, writeJSON, writeJSONSync } from './fileUtil'
+import shell from './shell'
+import uuidv4 from 'uuid/v4'
+import getSize from 'get-folder-size'
+
+/**
+ * Gets the total size (in bytes) of a given directory
+ * @param dirPath path to the directory for which to compute size
+ */
+const getSizeAsync: (dirPath: string) => Promise<number> = dirPath =>
```

Comment:
Treat this comment as informational, why did we have to read it into a `sync` block?